core/nightshadev2/src/fake_network.rs

```diff
@@ -20,6 +22,15 @@ fn spawn_all(num_authorities: usize) {
         let mut out_gossips_rx_vec = vec![];
         let mut consensus_rx_vec = vec![];
 
+        let mut public_keys = vec![];
+        let mut secret_keys = vec![];
+
+        for _ in 0..num_authorities {
```

Comment:
This can probably be rewritten as `let (public_keys, secret_keys): (Vec<_>, Vec<_>) = (0..num_authorities).map(|_| get_key_pair()).unzip();`

Reply:
Done